server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -206,13 +217,11 @@ private void stopNodeRecycleThread() {
     @Override
     public void afterSession(TestSession session) {
         LOGGER.entering();
-        synchronized (this) {
-            totalSessionsCompleted++;
-            if (totalSessionsCompleted <= maxSessionsAllowed) {
-                proxyLogger.fine("Completed session #" + totalSessionsCompleted + " (" + session.toString() + ")");
-            }
-            proxyLogger.fine("Total number of slots used: " + getTotalUsed() + " on node: " + getId());
+        totalSessionsCompleted++;
+        if (totalSessionsCompleted <= maxSessionsAllowed) {
```

Comment:
This `if` does not seem necessary. `totalSessionsCompleted` is not used to prevent new sessions when `maxSessionsAllowed` is reached in `getNewSession`